### PR TITLE
Drop down selection box

### PR DESCRIPTION
**Pull request by shusain**

### CODE REVIEW

**Comment by hassanbazzi** [comment]
font-awesome is an enormous library. We should be doing all this CSS / styling things inside main-app instead

File: src/dropDown.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        // otherwise open downward
+        function toggleDropDown(){
+          var rect = $element[0].getBoundingClientRect();
+          var middleOfWindow = window.innerHeight/2;
+          var middleOfElement = rect.top+rect.height/2;
+
+          if(middleOfElement>middleOfWindow){
+            $scope.direction = 'up';
+
+            dropDownListContainer[0].style.bottom = rect.height+'px';
```
Comment (hassanbazzi) [inline src/dropDown.js:182]: `bottom: 100%` achieves the same effect without calculation

**Comment by hassanbazzi** [comment]
Conflicts. Honestly, there's a ton of code in this. I feel it's a bit over-engineered for a dropdown menu but all the functionality is there. LGTM once conflicts are resolved.

**Comment by shusain** [comment]
Agree in general I wish it were less code and it needs to be better organized so it can be properly cleaned up.  That said I don't really know anything off hand that's in here that wasn't added out of necessity to keep all the current functionality.  Only bit of added functionality is for the keyboard support but it is not really a substantial part of the code.

On the plus side this should clean up a bit of the markup in the actual app since we'll just be defining the list element templates but the rest of the control is tucked away behind the directive.

Just saw there's an issue with the tooltips not cleaning up when switching states so going to get that fixed in another PR but will fix conflicts here first.

**Comment by shusain** [comment]
Just a note seems most of those conflicts were just because of rebuilding the dist folder every time, maybe good to figure out a plan for that so it's not always in conflict if two people are working or someone is working on more than one branch in here.

**Comment by shusain** [comment]
@hassanbazzi Okay to merge this?

**Comment by hassanbazzi** [comment]
LGTM

**Pull request merged**